### ISSUE TITLE
[BUG] Repeated use of `page` in CT results in `Cannot find parent object...`

### ISSUE THREAD

**Issue by hippyhorse** [microsoft/playwright#19206]
**Context:**
- Playwright Version: 1.26 and above (not present in 1.25.x)
- Operating System: Mac
- Node.js version: 18.12.1
- Browser: Chromium
- Extra: N/A

**Code Snippet**

Help us help you! Put down a short code snippet that illustrates your bug and
that we can run and debug locally. For example:

```javascript
// Seems to work consistently with '@playwright/test'
import { expect, test } from '@playwright/experimental-ct-react';

test.describe('Example', () => {
    // Mimic test data of n length
    const testData = new Array(100);

    for (let i = 0; i < testData.length; i++) {
        test(`Test ${i + 1}`, async ({ page }) => {
            await page.route('**/api/example', route => {
                route.fulfill({
                    body: JSON.stringify({ a: 1 })
                });
            });

            expect(true).toBeTruthy();
        });
    };
});
```

**Describe the bug**

Repeatedly accessing `page` (and seemingly `mount` too) results in the following error:

```
Error: Cannot find parent object browser-context@{guid} to create frame@{guid}
```

Observations:

- This only seems to be a problem with component testing (i.e. when `@playwright/experimental-ct-react` rather than `@playwright/test`)
- Does not appear to be an issue in `1.25.x`, but fails in all versions `1.26.x` and above

When [reading the docs](https://playwright.dev/docs/test-components#q-whats-the-difference-between-playwrighttest-and-playwrightexperimental-ct-reactsveltevuesolid) it mentions:

> Finally, under the hood, each test re-uses the `context` and `page` fixture as a speed optimization for Component Testing. It resets them in between each test so it should be functionally equivalent to `@playwright/test`'s guarantee that you get a new, isolated `context` and `page` fixture per-test.

However, it is not clear whether this differs `1.25.2` vs `1.26.0`, nor why it would cause the problem above.

In version `1.28.1` I get this error too (and it happens less frequently, so I sometimes just make the array length in the sample 200-300):

```
Pending operations:
  - browser._newContextForReuse
```

**Comment by mxschmitt** [comment]
I unfortunately can't repro, can you share maybe a repro in form of a Git repo?

**Comment by hippyhorse** [comment]
In the process of creating a fresh repo the ability to reproduce went away from me too. It would appear that there is somehow an issue in the code being bundled from my actual repo (i.e. the files being pulled in via vite)

I'll continue to try and narrow down the cause, but out of interest what would the usual cause of this be - something randomly causing the browser window to close? A long running process, something else?

**Comment by mxschmitt** [comment]
That should have nothing related to Playwright.

Over the past 2 years we have almost never received this error from our customers. Happy to dig into it if there is a specific repro for it.

I'll close it for now since there are no clear repro steps, but feel free to re-file if you have a repro, thanks!

**Comment by bradyisom** [comment]
@jm-mailosaur, did you ever figure out what was causing this issue in your environment? I am running into the same issue trying to run my component tests. It only seems to happen when there are multiple parallel workers running at the same time.

**Comment by jm-mailosaur** [comment]
@bradyisom we ended up disabling a whole set of tests in an area of the application we were due to re-write and that stopped other tests failing.

However we've new rewritten that area of the app - writing new tests as we go - and suddenly today have hit what seems to be a 'tippy point' where suddenly adding a new test (with no complex logic or loops) now causes the above error in a completely unrelated (and previously stable) test.

From what we can see, it appears to be Node running out of memory as increasing `--max-old-space-size` reduces the error count.

I think this should be re-opened, but its really hard to create reproduction steps without giving our actual tests over.

**Comment by jm-mailosaur** [comment]
@bradyisom **Update:** I believe the issue is the same as: https://github.com/microsoft/playwright/issues/6319

Give the following a try and let me know if it fixes it for you too. I've simply changed my test structure from:

```
test('a', ({ page }) => {
  // do test a
});

test('a', ({ page }) => {
  // do test b
});
```

to:

```
test.describe('A big group of tests', () => {
  // Ensure the page is closed after each test to avoid memory overload per https://github.com/microsoft/playwright/issues/6319
  test.afterEach(async ({ page }) => {
    await page.close();
  });
  
  test('a', ({ page }) => {
    // do test a
  });

  test('b', ({ page }) => {
    // do test b
  });
});
```

**Comment by mxschmitt** [comment]
If you use the page fixture, you don't have to close it manually, it gets done automatically for you, since the context gets closed.

If you have a repro, feel free to re-file an issue, we're more than happy to take a look. Thanks!

**Comment by jm-mailosaur** [comment]
@mxschmitt - totally get that it's supposed to do that, but https://github.com/microsoft/playwright/issues/6319 seemed to suggest that there is a memory leak causing `page.close` to be required manually.

However, I thought I'd double check as the other thing I changed was the addition of `test.describe`. So I did the following:

* Start with `test.describe` that has an `afterEach` block containing `page.close` call: **PASS**
* Removed the `test.describe` and by extension the `afterEach` block that contains the `page.close` call:  **FAIL**
* Reinstate `test.describe`, but not the `afterEach` block (just incase it was grouping the tests that fixed it):  **FAIL**
* Remove `test.describe` again, and instead add a single call to `await page.close()` at the end of every individual test: **PASS**

Based on the above it is definitely the addition of `page.close` that's fixed it for me (regardless of whether that should be necessary or not). I don't think this needs to be reopened as I'm of the opinion its the same issue underlying issue described in https://github.com/microsoft/playwright/issues/6319
